# Exploración de Datos — Precios de Propiedades en CABA

Dataset scrapeado de ZonaProp con propiedades en venta en 
Ciudad Autónoma de Buenos Aires.

## Objetivos
- Entender la estructura y calidad del dataset crudo
- Identificar nulos
- Ejemplos de cada columna, en que formato estan los strings
- Valores unicos
- Conclusiones

## Columnas
- `precio`: precio de venta en USD
- `expensas`: expensas mensuales en ARS
- `metros`: superficie total en m²
- `ambientes`: cantidad de ambientes
- `banos`: cantidad de baños
- `direccion`: dirección de la propiedad
- `barrio`: barrio y ciudad

### 1. Setup/Imports

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

### 2. Loading

In [3]:
df = pd.read_csv("../data/raw/zonaprop_raw.csv")

print(f"Filas: {len(df)}")
print(f"Columnas: {df.columns.tolist()}")

Filas: 38298
Columnas: ['precio', 'expensas', 'metros', 'ambientes', 'banos', 'direccion', 'barrio']


### 3. Exploración

In [4]:
df.head(10)

,precio,expensas,metros,ambientes,banos,direccion,barrio
0,USD 530.000,$ 770.000 Expensas,172 m² tot.,4 amb.,3 baños,Soldado De La Independencia 1200,"Las Cañitas, Palermo"
1,USD 170.000,NaN,73 m² tot.,4 amb.,1 baño,Malabia al 200,"Villa Crespo, Capital Federal"
2,USD 120.000,$ 260.000 Expensas,54 m² tot.,3 amb.,1 baño,Paraguay e/ Jean Jaures y Doctor Tomás Manuel ...,"Recoleta, Capital Federal"
3,USD 220.000,$ 380.073 Expensas,68 m² tot.,3 amb.,2 baños,PICO 1600. Entre Libertador y 11 De Septiembre,"Núñez, Capital Federal"
4,USD 2.250.000,$ 2.100.000 Expensas,399 m² tot.,4 amb.,3 baños,Av. Alvear 1500,"Recoleta, Capital Federal"
5,USD 84.100,$ 110.000 Expensas,55 m² tot.,1 amb.,1 baño,Av. Rivadavia al 9800,"Villa Luro, Capital Federal"
6,USD 550.000,$ 1 Expensas,723 m² tot.,NaN,6 baños,peru al 100,"Centro / Microcentro, Capital Federal"
7,USD 218.000,NaN,157 m² tot.,5 amb.,1 baño,Martiniano Leguizamón al 200,"Liniers, Capital Federal"
8,USD 105.000,$ 288.320 Expensas,57 m² tot.,3 amb.,1 baño,Muñiz 500,"Almagro, Capital Federal"
9,USD 173.251,NaN,52 m² tot.,1 amb.,1 baño,Weik Plaza - Esquina Juramento y Vuelta de Obl...,"Belgrano, Capital Federal"


#### 3.1 Dtypes

In [5]:
print("\nTipos de datos:")
print(df.dtypes)


Tipos de datos:
precio       str
expensas     str
metros       str
ambientes    str
banos        str
direccion    str
barrio       str
dtype: object


#### 3.2 Nulos

In [6]:
nulos = df.isnull().sum()
porcentaje = (nulos / len(df) * 100).round(2)

pd.DataFrame({
    "nulos": nulos,
    "porcentaje %": porcentaje
})

,nulos,porcentaje %
precio,0,0.00
expensas,16001,41.78
metros,18,0.05
ambientes,3287,8.58
banos,2655,6.93
direccion,0,0.00
barrio,0,0.00


#### 3.3 Ejemplos de cada columna

In [8]:
for col in df.columns:
    print(f"\n{'='*40}")
    print(f"  {col.upper()}")
    print(df[col].dropna().head(5).tolist())


  PRECIO
['USD 530.000', 'USD 170.000', 'USD 120.000', 'USD 220.000', 'USD 2.250.000']

  EXPENSAS
['$ 770.000 Expensas', '$ 260.000 Expensas', '$ 380.073 Expensas', '$ 2.100.000 Expensas', '$ 110.000 Expensas']

  METROS
['172 m² tot.', '73 m² tot.', '54 m² tot.', '68 m² tot.', '399 m² tot.']

  AMBIENTES
['4 amb.', '4 amb.', '3 amb.', '3 amb.', '4 amb.']

  BANOS
['3 baños', '1 baño', '1 baño', '2 baños', '3 baños']

  DIRECCION
['Soldado De La Independencia 1200', 'Malabia  al 200', 'Paraguay e/ Jean Jaures y Doctor Tomás Manuel de Anchorena', 'PICO 1600. Entre Libertador y 11 De Septiembre', 'Av. Alvear 1500']

  BARRIO
['Las Cañitas, Palermo', 'Villa Crespo, Capital Federal', 'Recoleta, Capital Federal', 'Núñez, Capital Federal', 'Recoleta, Capital Federal']


#### 4. Conclusiones

## Conclusiones - Qué hay que limpiar

### precio
- Formato: "USD 530.000"
- Limpiar: sacar "USD " y los puntos → int

### expensas
- Formato: "$ 770.000 Expensas"
- Limpiar: sacar "$ " y " Expensas" y los puntos → int
- 46% nulos → rellenar con 0

### metros
- Formato: "172 m² tot."
- Limpiar: sacar " m² tot." → int
- 18 nulos → eliminar filas

### ambientes
- Formato: "4 amb."
- Limpiar: sacar " amb." → int
- 9% nulos → analizar si imputar o eliminar

### banos
- Formato: "3 baños" o "1 baño"
- Limpiar: sacar " baños" y " baño" → int
- 7.5% nulos → analizar si imputar o eliminar

### direccion
- Formato inconsistente, no la usaremos en el modelo
- La dejamos como referencia

### barrio
- Formato: "Recoleta, Capital Federal" o "Las Cañitas, Palermo"
- Limpiar: quedarnos solo con el barrio principal
- Dos formatos distintos → hay que manejar ambos